src/USER-CONP/fix_charge_update.cpp

```diff
@@ -361,11 +379,11 @@ void FixChargeUpdate::create_taglist() {
 /* ---------------------------------------------------------------------- */
 
 std::vector<int> FixChargeUpdate::local_to_matrix() {
-  int const nmax = atom->nmax;
+  int const nall = atom->nlocal + atom->nghost;
```

Comment:
nmax ist genau atom->nlocal + atom->nghost. Brauchst du also nicht so machen. Mit nmax würde es mehr der LAMMPS Lingo entsprechen.

Reply:
Nein, nmax ist größer als nall. Wenn ich über nmax iteriere bekomme ich Atome mit tag = 0, die aber trotzdem in igroup sind.